src/main/java/seedu/address/model/Model.java

```diff
@@ -199,4 +200,29 @@ public interface Model {
      * @throws NullPointerException if {@code comparator} is null.
      */
     void sortBills(Comparator<Bill> comparator, boolean isAscending);
+
+    /**
+     * Updates previous state of the address book.
+     */
+    void updateAddressBookHistory();
+
+    /**
+     * Updates state of the address book when undo command is executed.
+     */
```

Comment:
redo or undo command is executed?

Reply:
when undo executed cos redo can only be done after undo is executed